src/commands/run.rs

```diff
@@ -81,6 +89,86 @@ pub async fn command(args: Args, _json: bool) -> Result<()> {
 
     ensure_project_and_environment_exist(&client, &configs, &linked_project).await?;
 
+    if args.remote {
+        let params = get_ssh_connect_params(
+            (
+                None, /* run doesn't have a project flag */
+                args.service.clone(),
+                args.environment.clone(),
+                None, /* default to latest deployment */
+            ),
+            &configs,
+            &client,
+        )
+        .await?;
+
+        let token = configs
+            .get_railway_auth_token()
+            .context("No authentication token found. Please login first with 'railway login'")?;
+
+        let spinner = create_spinner(true);
+
+        let ws_url = format!("wss://{}", configs.get_relay_host_path());
+        let mut terminal_client = establish_connection(&ws_url, &token, &params).await?;
+
+        // Run single command
+        let run_command = tokio::spawn(async move {
+            let e = execute_command(&mut terminal_client, args.args.clone(), spinner).await;
+            if let Err(e) = e {
+                bail!("Failed to execute command: {e:?}")
+            } else {
+                Ok(())
+            }
+        });
+        // Now, we need to:
+        // 1. Duplicate the current service
+        // 2. Disconnect the current service (the one the command is running on) from upstream
```

Comment:
I think we want to disconnect the new service from upstream?

Reply:
Nope, this isn't what Angelo described